cypress/e2e/tests/migration/migration-waves/sort.test.ts

```diff
@@ -0,0 +1,131 @@
+} from "../../../../utils/utils";
+import { startDate, endDate } from "../../../types/constants";
+
+import { MigrationWave } from "../../../models/migration/migration-waves/migration-wave";
+import { name } from "../../../types/constants";
+import { fieldHeader } from "../../../views/stakeholders.view";
+import { MigrationWaveView } from "../../../views/migration-wave.view";
+
+let migrationWavesList: Array<MigrationWave> = [];
+
+//Automates Polarion TC 341
+describe(["@tier2"], "Migration Waves sort validations", function () {
+    before("Login and Create Test Data", function () {
+        // Perform login
+        login();
+
+        // Create multiple Migration Waves
+        migrationWavesList = createMultipleMigrationWaves(2);
+    });
+    beforeEach("Persist session", function () {
+        // Save the session and token cookie for maintaining one login session
+
+        // Interceptors
+        cy.intercept("GET", "/hub/migration-waves*").as("getMigrationWaves");
+    });
+
+    after("Perform test data clean up", function () {
+        // Delete the Migration Waves created before the tests
+        deleteByList(migrationWavesList);
+    });
+
+    it("Name sort validations", function () {
+        // Navigate to Migration Waves tab
+        MigrationWave.open();
+        cy.get("@getMigrationWaves");
+        // get unsorted list when page loads
+        const unsortedList = getTableColumnData(name);
+
+        // Sort the Migration Waves by name in ascending order
+        sortAsc(name, MigrationWaveView.fieldHeader);
+        cy.wait(2000);
+
+        // Verify that the Migration Waves rows are displayed in ascending order
+        const afterAscSortList = getTableColumnData(name);
+        verifySortAsc(afterAscSortList, unsortedList);
+
+        // Sort the Migration Waves by name in descending order
+        sortDesc(name, MigrationWaveView.fieldHeader);
+        cy.wait(2000);
+
+        // Verify that the Migration Waves rows are displayed in descending order
+        const afterDescSortList = getTableColumnData(name);
+        verifySortDesc(afterDescSortList, unsortedList);
+    });
+
+    it("Start date sort validations", function () {
+        // Navigate to Migration Waves tab
+        MigrationWave.open();
+
+        // get unsorted list when page loads
+        const unsortedList = getTableColumnData(startDate);
+
+        // Sort the Migration Waves by Start date in ascending order
+        sortAsc(startDate, MigrationWaveView.fieldHeader);
+        cy.wait(2000);
+
+        // Verify that the Migration Waves rows are displayed in ascending order
+        const afterAscSortList = getTableColumnData(startDate);
+        verifyDateSortAsc(afterAscSortList, unsortedList);
+
+        // Sort the Migration Waves by Start date in descending order
+        sortDesc(startDate, MigrationWaveView.fieldHeader);
+        cy.wait(2000);
+
+        // Verify that the Migration Waves rows are displayed in descending order
```

Comment:
I'm not really a fan of commenting every line, especially if the previous test case does exactly the same and its already commented, but that's up to you.

Take a look at the clean code recommendation for comments to get more info https://gist.github.com/wojteklu/73c6914cc446146b8b533c0988cf8d29#comments-rules